web/src/views/CreateDandisetView/CreateDandisetView.vue

```diff
@@ -180,18 +243,69 @@ const name = ref('');
 const description = ref('');
 const license = ref<LicenseType>();
 const embargoed = ref(false);
+const hasAward = ref(true);
+const fundingSource = ref('');
 const awardNumber = ref('');
+const grantEndDate = ref('');
+
+// Calculate embargo end date as 2 years from today
+const embargoEndDate = computed(() => {
+  const today = new Date();
+  const twoYearsFromNow = new Date(today.getFullYear() + 2, today.getMonth(), today.getDate());
+  return twoYearsFromNow.toISOString().split('T')[0];
+});
+
+// Helper function to validate grant end date bounds
+const isGrantEndDateValid = computed(() => {
+  if (!grantEndDate.value) return false; // Required field
+
+  const selectedDate = new Date(grantEndDate.value);
+  const today = new Date();
+  const fiveYearsFromNow = new Date(today.getFullYear() + 5, today.getMonth(), today.getDate());
+
+  // Check if date is in the past or more than 5 years in the future
+  return selectedDate >= today && selectedDate <= fiveYearsFromNow;
+});
+
 const saveDisabled = computed(
   () => !name.value
       || !description.value
-      || (embargoed.value && !awardNumberValidator(awardNumber.value))
-      || (!embargoed.value && !license.value),
+      || (!embargoed.value && !license.value)
+      || (embargoed.value && hasAward.value && (!fundingSource.value || !grantEndDate.value || !isGrantEndDateValid.value))
+      || (embargoed.value && !hasAward.value && !embargoEndDate.value),
 );
 
 const awardNumberRules = computed(
-  () => [(v: string) => awardNumberValidator(v) || VALIDATION_FAIL_MESSAGE],
+  () => [(v: string) => !v || awardNumberValidator(v) || VALIDATION_FAIL_MESSAGE],
 );
 
+const fundingSourceRules = computed(
+  () => [(v: string) => !!v || 'Funding source is required'],
+);
+
+const grantEndDateRules = computed(() => [
+  (v: string) => !!v || 'Grant end date is required',
+  (v: string) => {
+    if (!v) return true; // Skip validation if empty (handled by required rule)
+
+    const selectedDate = new Date(v);
+    const today = new Date();
+    const fiveYearsFromNow = new Date(today.getFullYear() + 5, today.getMonth(), today.getDate());
```

Comment:
@mvandenburgh - the default duration is the end date of the grant if it's supported by nih, or a maximum of 2 years if not an nih award. if you want to simplify set it to 2 years by default if no grant end date is specified, not 5.